veloroutes_voies_vertes/actions.py

```diff
@@ -202,3 +202,71 @@ def update_selected_feature(*args):
             _add_selected_from_relation(agg_id, 'id_portion', 'segment', 'id_segment', 'element')
 
     layers[0].triggerRepaint()
+
+
+def create_relation(*args):
+    agg_id = int(args[0])
+    layer_name = args[1]
+    layer_update = args[2]
+
+    layers = QgsProject.instance().mapLayersByName(layer_name)
+
+    # Control if layers exists
+    if not len(layers):
+        iface.messageBar().pushCritical('Véloroutes', 'La couche {} n\'a pas été trouvée'.format(
+            layer_name))
+        return
+
+    src_layer = layers[0]
+
+    layers = QgsProject.instance().mapLayersByName(layer_update)
+
+    if not len(layers):
+        iface.messageBar().pushCritical('Véloroutes', 'La couche {} n\'a pas été trouvée'.format(
+            layer_update))
+        return
+
+    update_layer = layers[0]
+
+    # count number of features selected from src_layer
+    count = src_layer.selectedFeatureCount()
+    if count < 1:
+        iface.messageBar().pushCritical(
+            'Véloroutes',
+            'Vous devez sélectionner au moins un objet de la couche {}'.format(layer_update))
+        return
+
+    fields = update_layer.fields()
+    new_feature = QgsFeature()
+    new_feature.setFields(fields)
+
+    features = src_layer.getSelectedFeatures()
+    selected_features_id= []
+    for feat in features:
+        selected_features_id.append(feat['id_portion'])
+
+    update_layer.startEditing()
+
+    # open feature form and test if cancel form for stop or not the script
+    if not iface.openFeatureForm(update_layer, new_feature, showModal = True):
+        update_layer.rollBack()
+        iface.messageBar().pushInfo(
+                    'Véloroutes',
+                    'Création de l\'itinéraire annulée')
+        return
+    iface.messageBar().pushInfo(
+        'Véloroutes',
+        'Création de l\'itinéraire Validée')
+    update_layer.committedFeaturesAdded.connect(partial(_add_relation, selected_features_id))
+    update_layer.disconnect()
+    update_layer.commitChanges()
```

Comment:
```suggestion
    update_layer.committedFeaturesAdded.connect(partial(_add_relation, selected_features_id))
    update_layer.commitChanges()
    update_layer.committedFeaturesAdded.disconnect(partial(_add_relation, selected_features_id))
```
Faire la déconnection avec les mêmes paramètres après son utilisation.

C'est ainsi que c'est fait en C++ dans les tests de QGIS https://github.com/qgis/QGIS/blob/master/tests/src/core/testqgsvectorlayercache.cpp#L191

Reply:
En effet

Reply:
J'ai testé est cela retourne une erreur comme quoi l'objet n'est pas connecté. Et en laissant comme j'avais fais pas d'erreur